File: cz/mg/vulkan/vk/PFNvkGetDisplayPlaneCapabilitiesKHR.java
package cz.mg.vulkan.vk;

public class PFNvkGetDisplayPlaneCapabilitiesKHR extends VkFunctionPointer {
    public PFNvkGetDisplayPlaneCapabilitiesKHR() {
    }

    public PFNvkGetDisplayPlaneCapabilitiesKHR(VkMemory vkmemory) {
        super(vkmemory);
    }

    public PFNvkGetDisplayPlaneCapabilitiesKHR(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }


    public PFNvkGetDisplayPlaneCapabilitiesKHR(long value) {
        setValue(value);
    }

    public PFNvkGetDisplayPlaneCapabilitiesKHR(VkInstance instance) {
        super(instance, new VkString("vkGetDisplayPlaneCapabilitiesKHR"));
    }

    public void call(VkPhysicalDevice physicalDevice, VkDisplayModeKHR mode, VkUInt32 planeIndex, VkDisplayPlaneCapabilitiesKHR pCapabilities, VkResult rval){
        call(getValue(), physicalDevice != null ? physicalDevice.getVkAddress() : VkPointer.NULL_ADDRESS, mode != null ? mode.getVkAddress() : VkPointer.NULL_ADDRESS, planeIndex != null ? planeIndex.getVkAddress() : VkPointer.NULL_ADDRESS, pCapabilities != null ? pCapabilities.getVkAddress() : VkPointer.NULL, rval != null ? rval.getVkAddress() : VkPointer.NULL_ADDRESS);
    }

    private static native void call(long vkaddress, long physicalDevice, long mode, long planeIndex, long pCapabilities, long rval);
}

File: c/cz/mg/vulkan/PFN_vkGetDisplayPlaneCapabilitiesKHR.c
#include <vulkan/vulkan.h>
#include <jni.h>
#include <stdlib.h>
#include <string.h>

jlong jniPointerToLong(void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_vk_PFNvkGetDisplayPlaneCapabilitiesKHR_call(JNIEnv* env, jclass jc, jlong address, jlong physicalDevice, jlong mode, jlong planeIndex, jlong pCapabilities, jlong rval){
    (void)env;
    (void)jc;
    PFN_vkGetDisplayPlaneCapabilitiesKHR f = (PFN_vkGetDisplayPlaneCapabilitiesKHR)jniLongToFunctionPointer(address);
    VkResult* rvalAddress = jniLongToPointer(rval);
    *rvalAddress = f(
        *((VkPhysicalDevice*)jniLongToPointer(physicalDevice)),
        *((VkDisplayModeKHR*)jniLongToPointer(mode)),
        *((uint32_t*)jniLongToPointer(planeIndex)),
        ((VkDisplayPlaneCapabilitiesKHR*)jniLongToPointer(pCapabilities))
    );
}
